src/reducers/modules.js

```diff
@@ -55,7 +65,45 @@ const renameModuleState = (module, stateName, newName) => {
         }
       })
     }
+    switch(state['type']){
+      case 'Encounter':
+      case 'MedicationOrder':
+      case 'CarePlanStart':
+      case 'Procedure':
+        if(state.reason === stateName){
+          state.reason = newName;
+        }
+        break;
+      case 'ConditionOnset':
+      case 'AllergyOnset':
+        if(state.target_encounter === stateName){
+          state.target_encounter = newName;
+        }
+        break;
+      case 'Death':
+      case 'ConditionEnd':
+        if(state.condition_onset === stateName){
```

Comment:
There should also be an 'AllergyEnd' with an `allergy_onset` property to check here. I think this covers all the rest of them though.

I also think you may not need the switch at all, assuming checking `state.reason === stateName` won't crash if the state doesn't have a `reason` field. That way if we add new states with a `reason` field, or add a `reason` field to another existing state, we won't have to change this. We've been trying to be consistent such that the same field name means the same thing on all states, so a `reason` , `condition_onset`, etc, will always be a reference to another state.